librecomp/include/librecomp/overlays.hpp

```diff
@@ -5,24 +5,28 @@
 #include "sections.h"
 
 namespace recomp {
-    struct overlay_section_table_data_t {
-        SectionTableEntry* code_sections;
-        size_t num_code_sections;
-        size_t total_num_sections;
-    };
-
-    struct overlays_by_index_t {
-        int* table;
-        size_t len;
-    };
-
-    void register_overlays(const overlay_section_table_data_t& sections, const overlays_by_index_t& overlays);
-    void register_patch_section(SectionTableEntry* code_sections);
-    void load_patch_functions();
+    namespace overlays {
+        struct overlay_section_table_data_t {
+            SectionTableEntry* code_sections;
+            size_t num_code_sections;
+            size_t total_num_sections;
+        };
+
+        struct overlays_by_index_t {
+            int* table;
+            size_t len;
+        };
+
+        void register_overlays(const overlay_section_table_data_t& sections, const overlays_by_index_t& overlays);
+
+        void register_patch(const char* patch_data, std::size_t patch_size, SectionTableEntry* code_sections);
```

Comment:
This should be `register_patches`, since this data contains 1 or more patch functions and you can't really load two different patch files at once with how the system works currently.

Reply:
Oh. I was thinking on a patch as a set of functions, not like each function being an individual patch. I'll change it